src/fsharp/infos.fs

```diff
@@ -1276,8 +1289,23 @@ type MethInfo =
                      | _ -> ReflectedArgInfo.None
                  let isOutArg = (p.IsOut && not p.IsIn)
                  // Note: we get default argument values from VB and other .NET language metadata 
-                 let optArgInfo =  OptionalArgInfo.FromILParameter g amap m ilMethInfo.MetadataScope ilMethInfo.DeclaringTypeInst p 
-                 yield (isParamArrayArg, isOutArg, optArgInfo, reflArgInfo) ] ]
+                 let optArgInfo =  OptionalArgInfo.FromILParameter g amap m ilMethInfo.MetadataScope ilMethInfo.DeclaringTypeInst p
+
+                 let isCallerLineNumberArg = TryFindILAttribute g.attrib_CallerLineNumberAttribute p.CustomAttrs
+                 let isCallerFilePathArg = TryFindILAttribute g.attrib_CallerFilePathAttribute p.CustomAttrs
+
+                 let callerInfoInfo =
+                    match isCallerLineNumberArg, isCallerFilePathArg with
+                    | false, false -> NoCallerInfo
+                    | true, false -> CallerLineNumber
+                    | false, true -> CallerFilePath
+                    | true, true ->
+                        // if multiple caller info attributes are specified, pick the "wrong" one here
+                        // so that we get an error later
+                        if p.Type.TypeRef.FullName = "System.Int32" then CallerFilePath
+                        else CallerLineNumber
```

Comment:
@enricosada  The existing code is idiomatic given it's context in the compiler (and the compiler rule "code like the code around you")